src/Microsoft.ML.StandardLearners/Standard/LinearClassificationStatics.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Calibration;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Learners;
+using Microsoft.ML.StaticPipe.Runtime;
+
+namespace Microsoft.ML.StaticPipe
+{
+    using Arguments = StochasticGradientDescentClassificationTrainer.Arguments;
+
+    /// <summary>
+    /// Binary Classification trainer estimators.
+    /// </summary>
+    public static partial class BinaryClassificationTrainers
+    {
+        /// <summary>
+        ///  Predict a target using a linear binary classification model trained with the <see cref="Microsoft.ML.Runtime.Learners.StochasticGradientDescentClassificationTrainer"/> trainer.
+        /// </summary>
+        /// <param name="ctx">The binary classificaiton context trainer object.</param>
+        /// <param name="label">The name of the label column.</param>
+        /// <param name="features">The name of the feature column.</param>
+        /// <param name="weights">The name for the example weight column.</param>
+        /// <param name="maxIterations">The maximum number of iterations; set to 1 to simulate online learning.</param>
+        /// <param name="initLearningRate">The initial Learning Rate used by SGD.</param>
+        /// <param name="l2Weight">The L2 regularization constant.</param>
+        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TTupleInShape, TTupleOutShape, TTransformer}.Fit(DataView{TTupleInShape})"/> method is called on the
+        /// <see cref="Estimator{TTupleInShape, TTupleOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained.  Note that this action cannot change the result in any way; it is only a way for the caller to
+        /// be informed about what was learnt.</param>
+        /// <returns>The predicted output.</returns>
+        public static (Scalar<float> score, Scalar<float> probability, Scalar<bool> predictedLabel) StochasticGradientDescentClassificationTrainer(this BinaryClassificationContext.BinaryClassificationTrainers ctx,
+            Scalar<bool> label,
```

Comment:
>Scalar<bool> label, [](start = 12, length = 19)

Can you ask Pete or Tom about this one?
I'm just concern what we have Pisgty only on top of bool label. In other universe we used to support R4 where positive label was everything above zero, and negative is everything else and Key value with restriction to key dimensionality is 2.

Bool value is fine, but what if user have Text label, like "Yes" and "No", I don't think we have currently way to transform them to bool, only ToKey, but this method limit us only to bool.

Reply:
See CheckBinaryLabel method.

---
In reply to: [222534640](https://github.com/dotnet/machinelearning/pull/1134#discussion_r222534640) [](ancestors = 222534640)

Reply:
Ok I will check with them, and let you know. 

---
In reply to: [222534836](https://github.com/dotnet/machinelearning/pull/1134#discussion_r222534836) [](ancestors = 222534836,222534640)

Reply:
I checked with Tom and he said that in Pigsty it is important to be very deliberate about the types, and that we will just allow the labels to be Scalar<bool>. This is a constraint of the static api.

---
In reply to: [222775240](https://github.com/dotnet/machinelearning/pull/1134#discussion_r222775240) [](ancestors = 222775240,222534836,222534640)